packages/integrations/test/captureconsole.test.ts

```diff
@@ -0,0 +1,299 @@
+import { Event, Integration } from '@sentry/types';
+
+import { CaptureConsole } from '../src/captureconsole';
+
+const mockScope = {
+  setLevel: jest.fn(),
+  setExtra: jest.fn(),
+  addEventProcessor: jest.fn(),
+};
+
+const mockHub = {
+  withScope: jest.fn(callback => {
+    callback(mockScope);
+  }),
+  captureMessage: jest.fn(),
+  captureException: jest.fn(),
+};
+
+const getMockHubWithIntegration = (integration: Integration) => ({
+  ...mockHub,
+  getIntegration: jest.fn(() => integration),
+});
+
+// We're using this to un-monkey patch the console after each test.
+const originalConsole = Object.assign({}, global.console);
+
+describe('CaptureConsole setup', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+
+    // Un-monkey-patch the console functions
+    Object.assign(global.console, originalConsole);
+  });
+
+  it('should respect user-provided console levels', () => {
+    const captureConsoleIntegration = new CaptureConsole({ levels: ['log', 'warn'] });
+    captureConsoleIntegration.setupOnce(
+      () => undefined,
+      () => getMockHubWithIntegration(captureConsoleIntegration) as any,
+    );
+
+    expect(global.console.error).toBe(originalConsole.error); // not monkey patched
+    expect(global.console.log).not.toBe(originalConsole.log); // monkey patched
+    expect(global.console.warn).not.toBe(originalConsole.warn); // monkey patched
+  });
+
+  it('should fall back to default console levels if none are provided', () => {
+    const captureConsoleIntegration = new CaptureConsole();
+    captureConsoleIntegration.setupOnce(
+      () => undefined,
+      () => getMockHubWithIntegration(captureConsoleIntegration) as any,
+    );
+
+    // expect a set of defined console levels to have been monkey patched
+    expect(global.console.debug).not.toBe(originalConsole.debug);
+    expect(global.console.info).not.toBe(originalConsole.info);
+    expect(global.console.warn).not.toBe(originalConsole.warn);
+    expect(global.console.error).not.toBe(originalConsole.error);
+    expect(global.console.log).not.toBe(originalConsole.log);
+    expect(global.console.assert).not.toBe(originalConsole.assert);
+
+    // any other fields should not have been patched
+    expect(global.console.trace).toBe(originalConsole.trace);
```

Comment:
Yeah let's take this on afterwards! I'm gonna leave this test like this for now because we want to test the current behaviour. The PR that resolves #4599 should also adjust this test.